doc/Programming-guide/Key-platform-features/Common-Type-Library/index.md

```diff
@@ -71,80 +76,124 @@ Matching FQN in a different Tenant | N/A | OK | OK
+
+>**NOTE:** Kaa administrator creates new system CTs.
+>Tenant administrator creates new tenant CTs.
+>Tenant developer creates new application CTs.
+{:.note}
+
+In the **Add new type** window, fill in all the required fields and click **Add**.
 
-In **Add new type** window fill in all the required fields and click **Add** to apply the changes.
-Below is shown an example of **Add new type** window for Tenant CT.
 ![Create Tenant CTL](attach/create_tenant_ctl.png)
-There is also ability to create a new version of CT by clicking **Create new version** at the top of **CT details** window which is described in the next chapter.
 
-## Get or view CT details
+### View CT details
+
+To view the CT details:
+
+- Use the [REST API]({{root_url}}Programming-guide/Server-REST-APIs/#!/Common_Type_Library/getCTLSchemaById) call.
+
+OR
+
+- Open the **Administration UI** page, select the corresponding CTL and click on the CT in the list.
+The **Common type details** page will open.
 
-CT details can be obtained via [REST API]({{root_url}}Programming-guide/Server-REST-APIs/#!/Common_Type_Library/getCTLSchemaById) call or can be viewed in **CT details** window of Administration UI.
-To open CT details click the row of the CTs list in corresponding CTL window.
-Below is shown an example of **CT details** window for Application CT.
 ![View Application CTL](attach/view_application_ctl.png)
-To view another version of the CT, click on the version drop-down and select the appropriate version.
 
-## Delete CT schema
+To view another version of the CT, select it form the **Version** drop-down list.
+To create a new version of the CT, click **Create new version**.
+
+### Delete CT schemas
+
+To delete a schema:
 
-CT schema can be deleted via [REST API]({{root_url}}Programming-guide/Server-REST-APIs/#!/Common_Type_Library/deleteCTLSchemaByFqnVersionTenantIdAndApplicationToken) call or from **CT details** window of Administration UI.
-To delete CT schema open **CT details** and click **Delete** button located in the top of the window.
+- Use the [REST API]({{root_url}}Programming-guide/Server-REST-APIs/#!/Common_Type_Library/deleteCTLSchemaByFqnVersionTenantIdAndApplicationToken) call.
 
-> **NOTE:** The Kaa admin can delete System CT schema.
-The Tenant admin can delete Tenant CT schema.
-The Tenant developer can delete Application CT schema. 
+OR
 
-## CT promotion
+- Open the **Common type details** page and click **Delete**.
+
+>**NOTE:** Kaa administrator can delete system CT schemas.
+>Tenant administrator can delete tenant CT schemas.
+>Tenant developer can delete application CT schemas.
+{:.note}
+
+### Promote CTs
+
+If you want some of your **Application** scope CTs to be available in the **Tenant** scope, you neeed to promote them.
+To do this, you can use the [REST API]({{root_url}}Programming-guide/Server-REST-APIs/#!/Common_Type_Library/promoteScopeToTenant) call or use the **Promotion** feature:
```

Comment:
First, describe constraints (see [KAA-599](http://jira.kaaproject.org/browse/KAA-599)):

Users will be able to promote CT scope from APPLICATION to TENANT, provided that:
there is no other CT with identical FQN within the given tenant;
the user has the right to promote CTs;
the CT in question does not have dependencies on other CTs in the application scope.
Otherwise an appropriate error must be thrown. In case of dependencies on other CTs in the application scope, a list of such dependencies must be provided.